service/init.go

```diff
@@ -394,14 +461,18 @@ func (smf *SMF) SendNrfRegistration() {
 	//Check if another fresh NRF registration is required
 	if refreshNrfRegistration {
 		refreshNrfRegistration = false
-		if err := consumer.SendNFRegistration(); err != nil {
+		if prof, err := consumer.SendNFRegistration(); err != nil {
 			logger.InitLog.Infof("NRF Registration failure, %v", err.Error())
+		} else {
+			StopKeepAliveTimer()
+			StartKeepAliveTimer(prof)
```

Comment:
pass profile as pointer or pass only heartbeat timer value. Profile is big structure being passed and only heartbeat timer value is referred